openage/convert/terrain_convert.pyx

```diff
@@ -8,66 +8,89 @@ Transforms the dimetric tiles converted from SLP files to cartesian format.
 
 import math
 import argparse
-from PIL import Image
 from .slp import SLP
 from .texture import Texture
+from PIL import Image
+import numpy as np
 
 class Terrain_convert:
-
+    
+    # Method to simply put the tile on the big merge img. Something is wrong on this method because the result is transparent.
+    # TO FIX.
+    @staticmethod
+    def replace(overlayer, overlayed, overlayer_x, overlayer_y, overlayed_x, overlayed_y, slp_width, slp_height):
+        to_ret = overlayed
+        xr = overlayer_x
+        xd = overlayed_x
+        yr = overlayer_y
+        yd = overlayed_y
+        while xd < overlayed.shape[0]:
+            while yd < overlayed.shape[1]:
+                # if the overlayer is not transparent, then we can push it on the overlayed
+                # without the if, we would overwrite non-transparent pixels with transparent ones.
+                if xr < overlayer.shape[0] and yr < overlayer.shape[1] and overlayer[xr,yr][3]>0:
+                    to_ret[xd,yd] = overlayer[xr,yr]
+                yd+=1
+                yr+=1
+            xd+=1
+            xr+=1
+        return to_ret
+
+    
+    #Result of this method is transparent.
+    # TO FIX.
+    @staticmethod
     def merge(img):
         """
         This operation merges the single tiles together. The image converted
         from SLP contains 100 (10x10) different tiles for each terrain.
         """
 
         # Our merge result will be slightly smaller than the SLP image
-        merge_img = Image.new('RGBA', (962, 481), (0, 0, 0, 0))
-
-        # The size of each tile in the SLP
-        cdef int slp_tile_width = 98
-        cdef int slp_tile_height = 49
+        merge_img = np.empty([962,481,4], dtype=np.ndarray)
+        merge_img.fill(0)
```

Comment:
If you want the array to be filled with zeros from the start, you can call `numpy.zeros(..)` instead of `numpy.empty`.

Reply:
i'm gonna correct that, i'm not used to numpy.